src/CryptoDerivation.ts

```diff
@@ -30,30 +33,65 @@ export interface ICryptoDerivationStatic {
 }
 
 export class CryptoDerivation implements ICryptoDerivation {
-    public static async deriveKeyFromMaster(
-        masterKey: ICoreBuffer,
-        iterations: number,
-        keyAlgorithm: CryptoEncryptionAlgorithm,
-        salt?: ICoreBuffer
+    public static async deriveKeyFromPassword(
+        password: ICoreBuffer,
+        salt: ICoreBuffer,
+        keyAlgorithm: CryptoEncryptionAlgorithm = CryptoEncryptionAlgorithm.XCHACHA20_POLY1305,
+        derivationAlgorithm: CryptoDerivationAlgorithm = CryptoDerivationAlgorithm.ARGON2ID,
+        opslimit = 100000,
+        memlimit = 8192
     ): Promise<CryptoSecretKey> {
+        const sodium: any = (await SodiumWrapper.ready()) as any;
+        if (salt.buffer.byteLength !== 16) {
+            throw new Error(`The salt must be exactly ${sodium.crypto_pwhash_SALTBYTES} bytes long!`);
+        }
+        if (opslimit < sodium.crypto_pwhash_OPSLIMIT_MIN) {
+            throw new Error(`The opslimit must be higher than ${sodium.crypto_pwhash_OPSLIMIT_MIN}.`);
+        }
+        if (sodium.crypto_pwhash_OPSLIMIT_MAX > 0 && opslimit > sodium.crypto_pwhash_OPSLIMIT_MAX) {
+            throw new Error(`The opslimit must be lower than ${sodium.crypto_pwhash_OPSLIMIT_MAX}.`);
```

Comment:
How can that constant not be larger than 0?

Reply:
The default was -1 which means "no limit"